core/src/main/scala/org/http4s/Message.scala

```diff
@@ -221,10 +225,13 @@ case class Request(
 }
 
 object Request {
+
+  case class Connection(local: InetSocketAddress, remote: InetSocketAddress, secure: Boolean)
+
   object Keys {
     val PathInfoCaret = AttributeKey.http4s[Int]("request.pathInfoCaret")
     val PathTranslated = AttributeKey.http4s[File]("request.pathTranslated")
-    val Remote = AttributeKey.http4s[InetAddress]("request.remote")
+    val ConnectionInfo = AttributeKey.http4s[Connection]("request.remote")
```

Comment:
This gets modeled as an attribute rather than a property of the Request because it doesn't apply to the client?